Dockerfile

```diff
@@ -1,27 +1,26 @@
 FROM node:14-alpine AS BUILD
-COPY . /src
 
 # git is needed to install Half-Shot/slackdown
 RUN apk add git
 WORKDIR /src
+
+COPY package.json package-lock.json /src/
 RUN npm install
+COPY . /src
 RUN npm run build
 
 FROM node:14-alpine
 
 VOLUME /data/ /config/
 
-COPY package.json /usr/src/app/
-COPY package-lock.json /usr/src/app/
+WORKDIR /usr/src/app
+COPY package.json package-lock.json /usr/src/app/
+RUN apk add git && npm install --only=production
```

Comment:
I'm not sure how this makes things faster, could you explain how the ordering affects things?

Reply:
Same idea as before, we run `npm install` before copying the whole code, so if the code change but pacakge.json and package-lock.json don't docker can use the cache and we don't to have to run npm install again.